### PR TITLE
API/UCT: Added MD flags to indicate the need of local and remote keys.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by shamisp** [comment]
you also need to update xpmem

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3081/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/635/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2585/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2587/ for details (Mellanox internal link).

**Comment by alinask** [comment]
won't xpmem take the attributes of the mm md?

**Comment by shamisp** [comment]
I think you are right !

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3082/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2588/ for details (Mellanox internal link).

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -357,6 +357,11 @@ struct uct_iface_params {
 enum {
     UCT_MD_FLAG_ALLOC     = UCS_BIT(0),  /**< MD support memory allocation */
     UCT_MD_FLAG_REG       = UCS_BIT(1),  /**< MD support memory registration */
+    UCT_MD_FLAG_NEED_LKEY = UCS_BIT(2),  /**< The transport needs a valid local
+                                              memory key for zero-copy operations */
+    UCT_MD_FLAG_NEED_RKEY = UCS_BIT(3),  /**< The transport needs a valid
```
Comment (yosefe) [inline src/uct/api/uct.h]: let's rename this to UCT_MD_FLAG_NEED_MEMH (to be consistent in terminology)?

File: src/uct/sm/mm/mm_md.c [hunk 2]
```diff
@@ -119,6 +119,7 @@ ucs_status_t uct_mm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
         md_attr->reg_cost.overhead = 1000.0e-9;
         md_attr->reg_cost.growth   = 0.007e-9;
     }
+    md_attr->cap.flags |= UCT_MD_FLAG_NEED_RKEY;
```
Comment (yosefe) [inline src/uct/sm/mm/mm_md.c]: pls fix indentation on column

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3097/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2607/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/655/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2610/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2611/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@bbenton Can you please review as well?

**Comment by shamisp** [comment]
I don't see CMA code. Maybe I'm missing something ..

**Comment by alinask** [comment]
@shamisp Which of these flags does CMA need?

**Comment by bbenton** [comment]
:+1: from an API/Spec perspective

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/658/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/657/ for details.

**Comment by yosefe** [comment]
bot:bgateNEW:retest

**Comment by yosefe** [comment]
bot:bgateNEW:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/668/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3117/ for details.

**Comment by yosefe** [comment]
@shamisp pls take a look

**Comment by shamisp** [comment]
Do we need to set the flags for CMA:
```
md_attr->cap.flags         = UCT_MD_FLAG_REG |
                                 UCT_MD_FLAG_NEED_RKEY;
```

**Comment by yosefe** [comment]
@shamisp CMA does not need rkey

**Comment by shamisp** [comment]
We need to pass the PID, right ? I implemented it, but I have not touch it for a while ...

**Comment by yosefe** [comment]
Just PID and VA

**Comment by shamisp** [comment]
okay

**Comment by shamisp** [comment]
👍

**Pull request merged**